File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    cal = 0
    food_list = []
    food_list_reverse = []
    
    for i in range(1,len(food)):
        cal = (food[i] //2)
        food_list += str(i)*cal
        
    
    for i in food_list:
        if i == 3:
            food_list_revesre.insert(0,'0')
        food_list_reverse.append(str(int(i)))
        food_list_reverse.sort(reverse=True)
   
        answer += i 

    return answer+'0'+''.join(food_list_reverse)

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
import java.util.*;
class Solution {
    public String solution(int[] food) {
        String answer = "";
        int cal = 0;
        String a = "";
        List<String> food_list = new ArrayList<String>();
        StringBuilder f = new StringBuilder();
        StringBuilder b = new StringBuilder();
        
        for (int i = 1; i < food.length; i++){
            cal = food[i] / 2;
            for (int j =0; j < cal; j ++){
                f.append(i);
                b.append(i);
            }
            
        }
        // b = f;
        b.reverse();
        
        return f+"0"+b;
    }
}